Reference existing documentation

Lyla-Fischer opened the pull request:
Updated the README to refer to existing documentation, including the XBlock tutorial, external XBlocks, and sample XBlocks

edx-webhook commented:
Thanks for the pull request, @Lyla-Fischer! I've created [OSPR-907](https://openedx.atlassian.net/browse/OSPR-907) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).
We can't start reviewing your pull request until you've   added yourself to the [AUTHORS](https://github.com/edx/xblock-sdk/blob/master/AUTHORS) file . Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst) file for more information.

Lyla-Fischer commented:
I just noticed that I put one of my other pull requests as an ancestor of this pull request. It's a fairly uncontroversial pull request, though, and I can't figure out how to remove that particular commit from this pull request, so I'm hoping that we can just pretend like I opened this pull request after that uncontroversial one was accepted.

sarina commented:
Interactive rebase will allow you to remove commits. See https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes - once you're in interactive rebase mode, just remove from the .txt file the commit(s) that you don't want present on your branch.

Alternatively, make a new branch and cherry pick only the commits you want onto it.

Lyla-Fischer commented:
I just added some points to to my git-fu stats. :) Thanks!

Lyla-Fischer commented:
I think that my updated git-fu dropped the comments and tags that were on previous commits. I'll re-add the comments.

Lyla-Fischer commented on sample_xblocks/README.rst:
```diff
@@ -4,19 +4,32 @@ Demo XBlocks
+
+  * **basic.filethumbs:FileThumbsBlock** 
+
+Each of these blocks can be referenced in scenarios using a keyword in XML that is specified in the "entry_points" attribute of the setup tuple. 
```
@mhoeber My previous comment was dropped due to some rebasing, but I wanted to leave a record here that I asked you about proper documentation of these example classes. They could be useful as part of a tutorial, but quickly cause externally hosted documentation to become out of date as the examples mature.

Lyla-Fischer commented:
Tagging @nedbat for opinions

nedbat commented:
This is important, but it's going to be a few weeks before I can get to it.

nedbat commented on README.rst:
```diff
@@ -84,9 +84,9 @@ as the default does.
-views, and input handling.
+Making an XBlock involves creating a Python class that conforms to the XBlock 
+specification. See the sample_xblocks directory for examples and the 
+XBlock tutorial for a full walk-through: http://edx.readthedocs.org/projects/xblock-tutorial/en/latest/index.html
```
Please use ReST syntax for making links with English link text, and a URL behind the scenes:
```
See the sample_xblocks directory for examples and `the XBlock tutorial`_ for a full walk-through.

.. _the XBlock tutorial: http://edx.readthedocs.org/projects/xblock-tutorial
```

nedbat commented on README.rst:
```diff
@@ -110,13 +110,17 @@ own instance of Open edX, or released to the wider community).
 
-External XBlocks
+Example XBlocks
 ----------------
```
A minor nit, but please keep the punctuation line the same length as the section title. `</ocd>`

nedbat commented on README.rst:
```diff
@@ -110,13 +110,17 @@ own instance of Open edX, or released to the wider community).
-which have been installed and will appear in the workbench:
+Included in this repository are some example XBlocks that demonstrate how to use 
+various aspects of the XBlock SDK. You can see a more detailed description of 
+those examples in the readme located in that repository: https://github.com/edx/xblock-sdk/blob/master/sample_xblocks/README.rst
```
Another URL to rest-ify.

nedbat commented on README.rst:
```diff
@@ -110,13 +110,17 @@ own instance of Open edX, or released to the wider community).
+There is a rich community of XBlock developers that have put together a large 
+number of XBlocks that have been used in various contexts, mostly on the edx-platform. 
+You can see examples of what that community has done in the edx-platform wiki: 
+https://github.com/edx/edx-platform/wiki/List-of-XBlocks
```
Another URL to rest-ify.

nedbat commented on sample_xblocks/README.rst:
```diff
@@ -4,19 +4,32 @@ Demo XBlocks
-  * **view_counter:** A block which shows the total number of times it has been viewed
+  * **basic.problem:TextInputBlock** 
 
+  * **basic.problem:EqualityCheckerBlock** 
```
I'm wondering what to do with this block: it was a speculative design for how problems could be reimplemented as XBlocks, but we never used it, and likely never will convert capa to a new design like this.  I guess we should either remove it, or label it as "A possible design for how problems could be structured, but not used in edx-platform."

Lyla-Fischer replied:
I think that depends on whether we want to include examples of things that are not used in the edx-platform runtime. We don't currently have an alternative to show people how to build problems, however, I agree that the current level of exposure looks misleading to people using XBlock within edX's context.

Lyla-Fischer commented:
There was some confusing stuff with what was pulled into this pull request when I rebased it most recently. I fixed it, and the pull request makes sense again.

nedbat commented:
I've merged this, and will be making some further tweaks, coming to a PR near you.
